2-copy-of-code/lesson-14/amazon.html

```diff
@@ -45,7 +45,7 @@
 
         <a class="cart-link header-link" href="checkout.html">
           <img class="cart-icon" src="images/icons/cart-icon.png">
-          <div class="cart-quantity js-cart-quantity">0</div>
+          <div class="cart-quantity js-cart-quantity"></div>
```

Comment:
It still remains zero in cart instead of we can set cartQuantity = null;

Reply:
> It still remains zero in cart instead of we can set cartQuantity = null;

remove the '0' entirely and then try again. also run the function 'updateCartQuantity()' again at the beginning. That should fix it.

Reply:
i try both but still updatecartquantity() function make it 0 ☹️

Reply:
@vikrant-vikrant 
if (totalQuantity === 0 ) { document.querySelector('.js-cart-quantity').innerHTML = ''; } else { document.querySelector('.js-cart-quantity').innerHTML = totalQuantity; }

you can use this to fix it.

Reply:
it's too helpful brother thnx

Reply:
cartQuantity = "";  make it empty

Reply:
@vikrant-vikrant just saw your message and I was also struggling with this. What @Nishanthnaa52 said also works, but I found a shorter way to do it:
`
if (cartQuantity !== 0) {
    document.querySelector('.js-cart-quantity')
    .innerHTML = cartQuantity;
  }
`
You can simply place this inside the updateCartQuantity function. 

@Maheshgithub111 Btw, I gave a go to your code, but results in NaN when the cart is 0. And forces us to do some conversion, as cartQuantity = ""; is a string. It will basically show the combination of numbers.

Reply:
@ChickenWingBird, thanks for sharing your solution. I was beginning to panic that I'd done something wrong.

Reply:
function updateCartQuantity(){
        //updating cart element
        let cartQuantity = 0;

        cart.forEach((cartItem) => {
          cartQuantity += cartItem.quantity;
        });

        document.querySelector('.js-cart-quantity').
          innerHTML = cartQuantity||'';
    }
    
    what about this and reusing code again

Reply:
if(cartQuantity === 0){
        document.querySelector('.js-cart-quantity').innerHTML = `<div class="cart-quantity js-cart-quantity"></div>`
      }else{
        document.querySelector('.js-cart-quantity').innerHTML = `${cartQuantity}`;
      }

Reply:
> It still remains zero in cart instead of we can set cartQuantity = null;

function updateCartQuantityDOM() {
  const cartQuantity = calculateCartQuantity();
  const cartQuantityElement = document.querySelector('.js-cart-quantity');

  if (cartQuantity > 0) {
    cartQuantityElement.innerHTML = cartQuantity;
  } else {
    cartQuantityElement.innerHTML = '';

  }
}